app/types/type-parser.pegjs

```diff
@@ -0,0 +1,66 @@
+// This represents a scope
+// currently it doesn't make any sense to have a nested scope
+
+Definitions = _ "{" _ dfns:Definition* _ "}" _ { return new types.TypeScope(new Map(dfns)); }
```

Comment:
With this comment and any others I make, please let me know if I'm just misunderstanding this parser generator.